clang/employee-manager/src/handlers.c

```diff
@@ -1,32 +1,39 @@
 #include <libpq-fe.h>
-#include <string.h>
 #include <stdlib.h>
 
 #include "headers/handlers.h"
+
+#include "headers/db.h"
 #include "headers/employee.h"
 #include "headers/input.h"
 #include "headers/utils.h"
 #include "headers/screen.h"
 #include "headers/window.h"
 
 Employee* handlers_get_id(Window* win, Employee* employee) {
-  const char select_query[] = "SELECT * FROM employees WHERE id = $1 OR first = $1 OR last = $1";
-
   // get user input
   const char* const* user_input = input_get_search_input(win);
 
   // create query params from user input
   const char* const* query_params = user_input;
 
+  PGconn* conn = db_connect();
+
+  // query db with query params
+  PGresult* res = db_query_by_id(query_params);
+
   // get employee(s) data
-  employee_handle_search(query_params, employee);
+  employee = employee_convert(res, query_params, employee);
   if (!employee || employee == NULL) {
     printf("ERROR:: Failed to assign employee data to employee from get_employee in handlers_get_id\n");
```

Comment:
the error message is printing the wrong function name here

this should to be:
```c
printf("ERROR:: Failed to assign employee data to employee from employee_convert in handlers_get_id\n");
```

Reply:
![](https://media.tenor.com/images/b67356c0f095f0d1f51b3f355eebdf89/tenor.gif)